CMakeLists.txt

```diff
@@ -5,6 +5,8 @@ legacy_check()
 set(obs-websocket_VERSION 5.3.5)
 set(OBS_WEBSOCKET_RPC_VERSION 1)
 
+add_subdirectory(lib)
```

Comment:
Is there a specific reason you use `add_subdirectory` here? If you just wish to put the API stuff in a separate cmake file, you can use `include` (which also has the added benefit of not creating a new directory scope).

At this point in time it makes more sense to use an include and through the necessary code directly into the legacy script file, because there is no expectation of long-term maintainability of them (we'll probably remove the legacy stuff before that).

Reply:
I prefer subdirectory "isolation" over `include` that requires to twist your mind about the file is not "executed" where it is but where it is included.

Reply:
In this case it just creates lots of overhead - `add_directory` creates a new directory scope with all the baggage that comes with it, but you only use the subdirectory to house CMake-specific files.

Instead the public library is a "feature" of obs-websockets itself and thus should be integrated into it. Using an include just makes maintenance easier as all CMake code specific to the public API is contained in there, but that's a distinction for us as maintainers but irrelevant to CMake.

Reply:
Also this:
```
set_target_properties(
  obs-websocket-api
  PROPERTIES
             …
             PUBLIC_HEADER lib/obs-websocket-api.h)
```
Does not inspire me that the header will be without the `lib` folder once installed.

Reply:
`PUBLIC_HEADER` does not support directory structures - all specified files will be flattened into a single directory (which is why we needed to add extra code to maintain `libobs` custom subdirectory structure).

Reply:
Did changes, please review them.